openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -188,3 +202,24 @@ def process_pathways(self, site, pathways, programs):
                 logger.exception(u'Failed to process pathways for {domain}'.format(domain=site.domain))
                 failure = True
         return processed_pathways, programs, failure
+
+    def get_courses(self, programs):
+        """
+        Get all courses for the programs.
+
+        TODO: when course discovery can handle it, use that instead. That will allow us to put all course runs
+        in the cache not just the course runs in a program. Therefore, a cache miss would be different from a 
+        course not in a program.
+        """
+        course_runs = {}
+        failure = False
+
+        for program_uuid, program in programs.items():
```

Comment:
Maybe add a comment clarifying that these nested for loops are getting the courses via the program. It may also be more clean and clear style-wise if you use a mapping function instead of all of the nested for loops but that's my opinion.

Reply:
Stylistically, Python leans away from `map()` and `filter()` in favor of list or generator expressions as a community.

I tried to rewrite this as a chain of generators, but because the context you need at the bottom is the program_uuid from the top loop, it actually got fairly messy and unclear relative to the nested loops.